pennylane/workflow/execution.py

```diff
@@ -186,8 +186,15 @@ def cost_fn(params, x):
     if not tapes:
         return ()
 
-    ### Specifying and preprocessing variables ####
+    ### Apply the user transforms ####
+    user_post_processing = lambda x: x  # null basically
+    if transform_program and not transform_program.has_final_transform:
+        tapes, user_post_processing = transform_program(tapes)
+        if transform_program.is_informative:
+            return user_post_processing(tapes)
+        transform_program = TransformProgram()

```

Comment:
```suggestion
    transform_program = transform_program or TransformProgram:
    tapes, user_post_processing = transform_program(tapes)
    if transform_program.is_informative:
        return user_post_processing(tapes)
    transform_program = TransformProgram()

```

Reply:
`lambda` functions are kinda a pet peeve of mine given how hard they are to debug.